### ISSUE TITLE
Does reanimated support native animations of color and width? (Support Request)

### ISSUE THREAD

**Issue by AdamGerthel** [software-mansion/react-native-reanimated#588]
I have an existing, custom, progress bar component that I'm using in a project. It works the way I want it to using React Native Animated - except for the fact that it doesn't support `useNativeDriver: true` since I'm animating width and background color, so it's not as performant as I would've liked it to be. So, I turned to reanimated to search for a better solution.

The documentation states that reanimated supports the Animated API, but I'm having trouble understanding the API even though I've looked at tons of examples. I've used Animated quite extensively, but I'm still not able to grasp exactly how I switch to using reanimated. It's also not entirely clear to me whether I will be able to achieve my goal of improved performance.

So before I dive in and spend countless hours learning how to use reanimated:

1. Will I be able to improve the performance of my animated progress bar using reanimated?
2. Is the approach entirely different than it is when using Animated? (I would love to see a very simple before/after example to better understand the different approaches).

Any help is greatly appreciated!

This is my component:

```
import React from 'react'
import { View, Animated } from 'react-native'
import PropTypes from 'prop-types'

import style from './progress-bar.style'

class ProgressBar extends React.Component {
  constructor (props) {
    super (props)

    this.state = {
      value: new Animated.Value(props.value)
    }
  }

  UNSAFE_componentWillReceiveProps (nextProps) {
    const { value } = this.state

    if (nextProps.value !== value._value) {
      Animated.spring(value, {
        toValue: nextProps.value,
        bounciness: 2,
        velocity: 4
      }).start()
    }
  }

  shouldComponentUpdate (nextProps) {
    // Don't re-render unless max value changes since we're using Animated
    // to manage the value change and that will work without re-rendering.
    if (nextProps.maxValue !== this.props.maxValue) {
      return true
    } else {
      return false
    }
  }

  barColor = () => {
    const { value } = this.state
    const { maxValue, color } = this.props

    if (typeof color === 'object' && color.length === 2) {
      return value.interpolate({
        inputRange: [0, maxValue],
        outputRange: [color[0], color[1]]
      })
    } else {
      return color || style.progress.backgroundColor
    }
  }

  render () {
    const { value } = this.state
    const { maxValue, borderRadius, height } = this.props

    return (
      <View style={{...style.wrapper, borderRadius, height}}>
        <Animated.View
          style={[
            {
              ...style.progress,
              borderRadius,
              backgroundColor: this.barColor()
            },
            {
              width: value.interpolate({
                inputRange: [0, maxValue],
                outputRange: ['0%', '100%']
              })
            }
          ]}/>
      </View>
    )
  }
}

ProgressBar.defaultProps = {
  borderRadius: 10,
  height: 10
}

ProgressBar.propTypes = {
  color: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.string
  ]),
  value: PropTypes.number.isRequired,
  maxValue: PropTypes.number.isRequired,
  borderRadius: PropTypes.number
}

export default ProgressBar
```

**Comment by 4yki** [comment]
Hi Adam! According to your ref, "backgroundColor" is a fully supported style prop. 
Talking about "width" I'd suggest you to play with scaleX and translationX of "transform" prop. It will definitely work better than width interpolation.

**Comment by AdamGerthel** [comment]
@4yki Alright, good to know! Yes, I have a working solution using translation instead of width (overflow: hidden on the parent and move the child out of the visible view) so I can solve that part.

So in summary, if I move to using reanimated instead of using Animated I should be able to gain performance improvements vs my current solution?

I tried replacing Animated with Animated from Reanimated (since the same API is supposed to be supported) but I got an error from my color interpolation that I'm got getting using Animated from react-native.

Do I need to rewrite the component entirely to make use of reanimated? I looked at the [example in the docs](https://software-mansion.github.io/react-native-reanimated/declarative-animation-api.html) in order to better understand how to migrate from Animated to Reanimated, but the example components there aren't the same, so it's hard to compare what they do (it's also not clear what the components actually attempt to *do*, the docs just say "...a component that renders").
